pkg/controllers/kubectl_delivery/controller.go

```diff
@@ -302,3 +301,24 @@ func (c *KubectlDeliveryController) enqueue(obj interface{}) {
 	}
 	c.queue.AddRateLimited(key)
 }
+
+
+// IsPodReadyConditionTrue returns true if a pod is ready; false otherwise.
+func isPodReadyConditionTrue(status corev1.PodStatus) bool {
+	_, condition := getPodConditionFromList(status.Conditions, corev1.PodReady)
+	return condition != nil && condition.Status == corev1.ConditionTrue
+}
+
+// GetPodConditionFromList extracts the provided condition from the given list of condition and
+// returns the index of the condition and the condition. Returns -1 and nil if the condition is not present.
+func getPodConditionFromList(conditions []corev1.PodCondition, conditionType corev1.PodConditionType) (int, *corev1.PodCondition) {
+	if conditions == nil {
+		return -1, nil
+	}
+	for i := range conditions {
+		if conditions[i].Type == conditionType {
+			return i, &conditions[i]
+		}
+	}
+	return -1, nil
+}
```

Comment:
Please gofmt the file. There should be a new line.

Reply:
Sorry I forgot it. Now I have gofmt this file and squash few commit.